backend/metadata_writer/src/metadata_writer.py

```diff
@@ -196,15 +196,15 @@ def get_output_template(obj):
 pods_with_written_metadata = set()
 
 while True:
-    print("Start watching Kubernetes Pods created by Argo")
-    for event in k8s_watch.stream(
-        k8s_api.list_namespaced_pod,
-        namespace=namespace_to_watch,
-        label_selector=PIPELINE_LABEL_KEY,
-        timeout_seconds=1800,  # Sometimes watch gets stuck
-        _request_timeout=2000,  # Sometimes HTTP GET gets stuck
-    ):
-        try:
+    print("Start watching Kubernetes Pods created by Argo or Tekton")
+    try:
+        for event in k8s_watch.stream(
+            k8s_api.list_namespaced_pod,
+            namespace=namespace_to_watch,
+            label_selector=PIPELINE_LABEL_KEY,
+            timeout_seconds=1800,  # Sometimes watch gets stuck
+            _request_timeout=2000,  # Sometimes HTTP GET gets stuck
```

Comment:
what will happen in event of timeout?

Reply:
In the original version it will just crash and restart. This happens quite often as the numbers of pod add up. Moving the `try` a level above will just log this error without crashing the pod itself.